src/authz/data/apiHooks.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from '@tanstack/react-query';
+import { PermissionValidationRequest, PermissionValidationResponse } from '@src/authz/types';
+import { validateUserPermissions } from './api';
+
+const adminConsoleQueryKeys = {
+  all: ['authz'],
+  permissions: (permissions: PermissionValidationRequest[]) => [...adminConsoleQueryKeys.all, 'validatePermissions', permissions] as const,
+};
+
+/**
+ * React Query hook to validate if the current user has permissions over a certain object in the instance.
+ * It helps to:
+ * - Determine whether the current user can access certain object.
+ * - Provide role-based rendering logic for UI components.
+ *
+ * @param permissions - The array of objects and actions to validate.
+ *
+ * @example
+ * const { data } = useValidateUserPermissions([{
+           "action": "act:read",
+           "scope": "org:OpenedX"
+       }]);
+ * if (data[0].allowed) { ... }
+ *
+ */
+export const useValidateUserPermissions = (
```

Comment:
This is minor/optional feedback about the name of this hook:

To me, "validate user permissions" sounds like an action, like it would throw an exception if the user doesn't have some permissions. But this is just fetching some data, not making an action.

I think "useScopedPermissions" or just "useUserPermissions" or something like that would better reflect that this is just getting the user permissions, but you still have to validate/check that they're `allowed` or not yourself.